runtime/containers.go

```diff
@@ -15,6 +16,7 @@ import (
 )
 
 type ContainerTarget struct {
+	ctx context.Context
```

Comment:
Don't store `Context` as a struct field: pass it as a parameter wherever possible. https://github.com/golang/go/wiki/CodeReviewComments#contexts

The main exceptions are:
1. Satisfying an interface that doesn't take in a `Context`
2. Designing code that creates truly asynchronous work like in buildagent.

Passing the `Context` as the first parameter helps the caller arrange for timeouts/cancellation for specific operations, if needed.